### DISCUSSION TITLE
Why is defining a detect role recommended, even if I use for recording only?

### DISCUSSION THREAD

**Discussion by akcasoy** [blakeblackshear/frigate#14574] [Ask A Question] [answered]
Can somebody please explain me this section of the documentation (written in camera configuration Page of documentation) why is this recommended?:

<img width="883" alt="Bildschirmfoto 2024-10-25 um 11 38 55" src="https://github.com/user-attachments/assets/a1be9eda-18b9-4fa0-9f47-27edd85f8d24">

Is this suggestion independent of this flag?

 ```yaml
 detect:
    enabled: false
```

I mean even if i disabled the detection, should i put a substream for detection? (and why?)
Or does it mean even if we just want to record, we should also enable detection and exclude the camera in this way:

<img width="921" alt="Bildschirmfoto 2024-10-25 um 11 47 43" src="https://github.com/user-attachments/assets/0fea47bf-3607-4de5-b50e-d9696a33ac7f">

**Comment by hawkeye217** [comment, marked as answer]
Frigate still decodes and uses the stream you've defined for the detect role for other features, even if you have disabled object detection. If you use a substream from your camera for this role, it minimizes additional resource usage. 

The reviews detections features are for categorization of your footage and are unrelated to enabling/disabling object detection.
